Updated chapter02.rst for Django 1.4.1

lgp171188 opened the pull request:
Made changes for the Django 1.4.1 project layout change.

lgp171188 committed:
Updated chapter02.rst for Django 1.4.1

Made changes for the Django 1.4.1 project layout change.

jacobian commented on chapter02.rst:
```diff
@@ -399,29 +399,47 @@ To do this, navigate to the directory where ``django-admin.py`` is installed
+    The default project layout recently changed. If you're seeing a 
+    "flat" layout (with no inner ``mysite/`` directory), you're probably using
+    a version of Django that doesn't match this tutorial version. 
+    You'll want to switch to the newer Django version.
```
I find "You'll want to switch to the newer Django version." rather glib -- perhaps they _don't_ want to switch (or can't). Perhaps instead replace this with a pointer to older documentation? Something like "This book covers Django 1.4 and above, so if you're using an older version you probably want to consult Django's official documentation: [link to 1.3], [link to 1.2], etc."

lgp171188 replied:
Once this book is updated to the latest and greatest Django version, will the old versions be still available on djangobook.com? If so, it'd be better to link to them. If not, we could link to the official Django documentation at least.

jacobian commented on chapter02.rst:
```diff
@@ -399,29 +399,47 @@ To do this, navigate to the directory where ``django-admin.py`` is installed
+  "table of contents" of your Django-powered site.
+
+* ``wsgi.py``: An entry-point for WSGI-compatible webservers to serve your project. 
+  See How to deploy with WSGI for more details.
```
Is this section in the book? If so this should be a link to the book. Otherwise, it should be a link to the docs.

lgp171188 replied:
My bad. I copied the description from the official Django documentation and forgot to link 'How to deploy with WSGI' . I will make the changes and update the pull request.

lgp171188 commented:
Is it a good idea to complete the whole update to Django 1.4.1 in a branch and merge it, or is it okay to do it in small parts like I have done now?

lgp171188 commented:
Also the python version and compatibility details need to be updated. I understand that Django 1.4 supports Python 2.5 to 2.7. Not sure about GIS framework.

lgp171188 committed:
Incorporated Jacob's comments and more 1.4 changes

Incorporated Jacob's comments. Updated the 'Installing the "development" version' section to use Git as Django is versioned on Git and a few more 1.4 changes.

lgp171188 commented:
Hi Jacob, I have incorporated the changes you suggested. Also made some changes for the Python version compatibility and Git instead of subversion for installing the "development" version. I am not sure if "master" branch needs to be used for the development version.
